File: LeetCode_Algorithm/520(Detect Capital)/520.py
class Solution:
    # 23ms 99% 13.9MB 56%
    def detectCapitalUse(self, word: str) -> bool:
        if len(word) == 1:
            return True
        if not self.isCapital(word[0]) and self.isCapital(word[1]):
            return False
        capital = self.isCapital(word[1])
        for i in range(2, len(word)):
            if capital != self.isCapital(word[i]):
                return False
        return True
    def isCapital(self, c):
        return c >= 'A' and c <= 'Z'
print(Solution().detectCapitalUse("Leet"))

File: LeetCode_Algorithm/520(Detect Capital)/520.js
isCapital = (letter) => {
  return (letter >= 'A') && (letter <= 'Z')
}
var detectCapitalUse = function(word) {
  // sol1  80ms 65% 42.2MB 60%
  if (word.length == 1)
    return true
  if (!isCapital(word[0]) && isCapital(word[1]))
    return false
  capital = isCapital(word[1])
  for (let i=2; i < word.length; i++) {
    if (capital != isCapital(word[2]))
      return false
  }
  return true
};
console.log(detectCapitalUse("Leet"))
console.log(detectCapitalUse("lEet"))